tests/dir.sdkbuild.props

```diff
@@ -0,0 +1,25 @@
+<Project>
+  <Import Project="../dir.common.props" />
+
+  <!-- This file contains build properties that apply to every SDK
+       project under the test tree, and also generated test projects
+       in bin/tests. Eventually every test project should build using
+       the SDK. -->
+
+  <PropertyGroup>
+    <OSPlatformConfig>$(BuildOS).$(Platform).$(Configuration)</OSPlatformConfig>
+
+    <TestSrcDir>$(CoreclrDir)/tests/src</TestSrcDir>
+    <BuildProjectRelativeDir>$([System.String]::Copy('$(MSBuildProjectDirectary)').Replace($(TestSrcDir),''))/$(MSBuildProjectName)</BuildProjectRelativeDir>
+
+    <!-- BaseIntermediateOutputPath is used by the SDK as the location
+         for the lock file, and props/targets from nuget packages. -->
+    <BaseIntermediateOutputPath>$(CoreclrDir)/bin/tests/obj/$(OSPlatformConfig)/Managed/$(BuildProjectRelativeDir)</BaseIntermediateOutputPath>
```

Comment:
Can we use `RootBinDir` here?

Reply:
They're on different import chains. :) coreclr/dir.props has a different `BaseIntermediateOutputPath` and a bunch of other properties that the SDK test projects don't need. `BaseIntermediateOutputPath` for SDK projects needs to be different because it's used in a very specific way by the SDK. See this: https://github.com/Microsoft/msbuild/issues/1603#issuecomment-341116396.